gcp/live/prd/k8s/gpii/couchdb-prometheus-exporter/terraform.tfvars

```diff
@@ -0,0 +1,24 @@
+# ↓ Module metadata
+terragrunt = {
+  terraform {
+    source = "/project/modules//couchdb-prometheus-exporter"
+  }
+
+  dependencies {
+    paths = [
+      "../couchdb",
+    ]
+  }
+
+  include = {
+    path = "${find_in_parent_folders()}"
+  }
+}
+
+# ↓ Module configuration (empty means all default)
+
+replica_count     = 2
```

Comment:
I was thinking of [Prometheus's "High Availability" strategy](https://prometheus.io/docs/introduction/faq/#can-prometheus-be-made-highly-available):  scrape it N times and let someone downstream de-dupe it.

The incoming metrics are tagged with the Node running the metrics exporter. I think that should be sufficient if we need to de-dupe or otherwise distinguish which exporter produced a specific metric. (I think I will add some scheduling hints to prevent two exporter Pods from running on the same Node.)

The advantage of this system is simplicity: we won't lose all metrics collection during Pod or Node restarts, but the exporters don't have to know anything about one another or coordinate in any way.

Reply:
Where will the de-duping happen?

Won't we see all the metrics twice in Stackdriver? This sounds a bit in opposition with the concern of not shipping all/too many things to Stackdriver (as we have to pay for that). I would personally wonder if it won't be too confusing seeing all the metrics 2x (if that's the case).

Otherwise, 👍 for thinking about HA, I agree that from that point of view this seems simple & good. Just need to make the de-duping happen somewhere imho.

Reply:
> This sounds a bit in opposition with the concern of not shipping all/too many things to Stackdriver (as we have to pay for that).

I don't mind paying for HA. What I don't like is paying for metrics we will definitely never use.

>  I would personally wonder if it won't be too confusing seeing all the metrics 2x (if that's the case).

In my limited experience with Stackdriver Metrics Explorer and Dashboard so far, I have not found it confusing. The incoming metrics are tagged with the Node running the metrics exporter. I think that should be sufficient if we need to de-dupe or otherwise distinguish which exporter produced a specific metric.

Do you have an alternative solution for HA that avoids these issues?

Reply:
Conversation continues [below](https://github.com/gpii-ops/gpii-infra/pull/230#pullrequestreview-184729878).